lms/templates/courseware/courseware.html

```diff
@@ -81,6 +81,10 @@
     BookmarksFactory();
   </%static:require_module>
 
+  <%static:require_module module_name="js/courseware/link_clicked_events" class_name="LinkClickedEvents">
```

Comment:
This is adding a new JavaScript file to be loaded on every courseware page which is a performance issue even though it is a small file. 

It would be better if we could bundle it into an existing file. I wonder if we should rename the ``BookmarksFactory`` as something like ``CoursewareFactory``, and then have that depend upon the various event files. In an ideal world, all the JS would be loaded by a single per-page bundle, so calling it ``CoursewareFactory`` makes some sense to me. You could also introduce `CoursewareFactory`` in addition to ``BookmarkFactory``, and then it would invoke all of the dependent factories.

I would suggest checking in with @tobz  to see if he has a recommendation for how to manage this.